src/Esprima/Utils/AstJson.cs

```diff
@@ -146,7 +146,7 @@ public static void WriteJson(this Node node, JsonWriter writer, Options options)
             visitor.Visit(node);
         }
 
-        private sealed class Visitor : AstVisitor
+        private sealed partial class Visitor : AstVisitor
```

Comment:
would it work here if it would be `JsxVisitor : Visitor` and let Jsx AST take `JsxVisitor`?